src/solvers/smt2/smt2_tokenizer.h

```diff
@@ -56,6 +64,19 @@ class smt2_tokenizert:public parsert
     return messaget::error();
   }
 
+  /// skip any tokens until all parentheses are closed
+  /// or the end of file is reached
+  void skip_to_end_of_list();
+
+  struct smt2_errort : public cprover_exception_baset
+  {
+    /// this is a dummy -- errors are reported via the error() stream
+    std::string what() const override
+    {
+      return "";
```

Comment:
This is a bit of non-standard use of exceptions. The error message should be attached to it so that the fatal error handling and printing can be done by the driver.

Reply:
Ok, now with error message in the exception.